src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1694,15 +1696,30 @@ def get_disk_driver(self) -> Optional[ManagedClusterStorageProfileDiskCSIDriver]
                 "--disable-disk-driver at the same time."
             )
 
+        if disable_disk_driver and azuredisk_csi_version:
+            raise ArgumentUsageError(
+                "The parameter --disable-disk-driver cannot be used "
+                "when --azuredisk-csi-version is specified.")
+
+        # Setting default driver as v1 if not mentioned
+        if not azuredisk_csi_version:
+            azuredisk_csi_version = "v1"
+
+        if not (azuredisk_csi_version == "v1" or azuredisk_csi_version == "v2"):
+            raise InvalidArgumentValueError(
+                "Valid values for --azuredisk-csi-version are v1 and v2")
```

Comment:
For enum-like validation, please take [network plugin](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/_params.py#L198) as the example and perform the validation at parameter declaration.